app/templates/views/re-validate-email-sent.html

```diff
@@ -11,7 +11,7 @@
   <div class="govuk-grid-column-two-thirds">
     <h1 class="font-body-2xl margin-bottom-3">{{ title }}</h1>
     <p>For security, we need to check if you still have access to your email address.</p>
-    <p>We’ve sent you a link to sign in to Notify. The link will open in a new browser window, so you can close this one.</p>
+    <p>We’ve sent you a link valid for 1 hour to sign in to Notify. The link will open in a new browser window, so you can close this one.</p>
```

Comment:
Is the link valid for just 1 hour?  I don't remember seeing that, but that's not to say it isn't true, either. 😄

Reply:
Sure thing, thanks for pointing me to both of those!

Reply:
I did some digging, here's what I uncovered:

`EMAIL_EXPIRY_SECONDS` is used in the `app/main/views/verify.py` view, specifically in the [`verify_email`](https://github.com/GSA/notifications-admin/blob/main/app/main/views/verify.py#L39-L68):

```python
token_data = check_token(
    token,
    current_app.config['SECRET_KEY'],
    current_app.config['DANGEROUS_SALT'],
    current_app.config['EMAIL_EXPIRY_SECONDS']
)
```

Meanwhile, `EMAIL_2FA_EXPIRY_SECONDS` is used in `app/main/views/two_factor.py`, specifically with the [`two_factor_email`](https://github.com/GSA/notifications-admin/blob/main/app/main/views/two_factor.py#L43-L66) method:

```python
token_data = json.loads(check_token(
    token,
    current_app.config['SECRET_KEY'],
    current_app.config['DANGEROUS_SALT'],
    current_app.config['EMAIL_2FA_EXPIRY_SECONDS']
))

Ultimately, which config variable gets used depends on which route is hit:

- `/email-auth/<token>` = `two_factor.py` view
- `/verify-email/<token>` = `verify.py` view

I think you're right in that it's the `/email-auth/<token>` route for this flow based on what we saw, which means it's actually going to check against the 30 minute config (`EMAIL_2FA_EXPIRY_SECONDS`).

You could test this by doing the following, or tweaking one of the tests to do the following:

- Send yourself a new verification email
- Update it's time (likely `created_at`) to be more than 30 minutes ago but less than an hour
- Try to verify your email

If the verification fails, then we know for sure it's saying one thing and doing another. 😆 

In that case, I think it's safe to say we can change the config variable used in the `two_factor_email` method in `two_factor.py` to be `EMAIL_EXPIRY_SECONDS` instead and match and go from there.

As an aside, it seems odd to me that these things are interwoven as they are and that things don't just flow through the verify email methods that exist...  I'm not sure why `two_factor_email` is fitting into verifying your email instead of something like just during a login sequence with 2FA...

Reply:
I replaced the variable in the logic to match the 1 hour variable set in the config. I ran some tests locally changing expiration values to make sure it works as intended.